app/src/main/java/com/yoshitoke/weatheringwithyou/mvp/Contract.kt

```diff
@@ -0,0 +1,39 @@
+package com.yoshitoke.weatheringwithyou.mvp
+
+import com.yoshitoke.weatheringwithyou.mvp.model.DataClass.WeatherInfo
+
+interface Contract {
+    interface View {
+
+        fun showSpinnerList(data: List<String>)
+
+        fun showWeatherInfo(data: WeatherInfo)
+
+    }
+
+    interface Model {
+
+        interface OnWeatherFinishedListener {
+            fun onFinished(data: WeatherInfo?)
+        }
+
+        interface OnCityListFinishedListener {
+            fun onFinished(cityListName: List<String>)
+        }
+
+        interface OnFetchErrorListener {
+            fun onError(error: String?)
+        }
+
+    }
+
+    interface Presenter {
+
+        fun loadCityList()
+
+        fun onLocationSwitched(listPosition: Int)
+
+        fun destroyView()
+
+    }
```

Comment:
Create new interface `BasePresenter`, define `fun destroy()` here.
interface `Presenter` implement `BasePresenter`
`destroy` is enough, `destroyView` seems to be redundant

Reply:
what benefits will BasePresenter provide sir?

Reply:
Just my suggestion, it's necessary to develop more features. You can improve later